source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -432,12 +430,18 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
   }
 }
 
-bool Filter::isBufferFull() const {
+bool Filter::isBufferFull(uint64_t num_bytes_processing) const {
+  if (!config_->allowPartialMessage()) {
+    return false;
+  }
+
+  uint64_t num_bytes_buffered = num_bytes_processing;
   const auto* buffer = decoder_callbacks_->decodingBuffer();
-  if (config_->allowPartialMessage() && buffer != nullptr) {
-    return buffer->length() >= config_->maxRequestBytes();
+  if (buffer != nullptr) {
+    num_bytes_buffered += buffer->length();
   }
-  return false;
+
+  return num_bytes_buffered >= config_->maxRequestBytes();
 }
```

Comment:
The decoding buffer only has the data that's already been processed by `decodeData()`. We want to know whether the total amount of data received (previously received plus currently processing) is enough to fill the auth request.